packages/ai-chat/src/browser/file-chat-variable-contribution.ts

```diff
@@ -216,13 +216,17 @@ export class FileChatVariableContribution implements FrontendVariableContributio
             const nodes: string[] = JSON.parse(data);
             const variables: AIVariableResolutionRequest[] = [];
             const texts: string[] = [];
+            const rootPaths = (await this.wsService.roots).map(root => root.resource.path.toString());
 
             for (const node of nodes) {
-                const [, filePath] = node.split(':');
-                if (!filePath) {
+                const rootPath = rootPaths.find(path => node.startsWith(path + ':'));
+
+                if (!rootPath) {
                     continue;
                 }
```

Comment:
Please add a comment on why we are doing this. When stumbling over it later on, developers will ask themselves why this is necessary.